py/server/deephaven/parquet.py

```diff
@@ -234,20 +248,121 @@ def write(
+            includes any non-existing directories, they are created. If there is an error, any intermediate directories
             previously created are removed; note this makes this method unsafe for concurrent use
-        col_definitions (Optional[List[Column]]): the column definitions to use, default is None
-        col_instructions (Optional[List[ColumnInstruction]]): instructions for customizations while writing, default is None
-        compression_codec_name (Optional[str]): the default compression codec to use, if not specified, defaults to SNAPPY
-        max_dictionary_keys (Optional[int]): the maximum dictionary keys allowed, if not specified, defaults to 2^20 (1,048,576)
-        max_dictionary_size (Optional[int]): the maximum dictionary size (in bytes) allowed, defaults to 2^20 (1,048,576)
+        col_definitions (Optional[List[Column]]): the column definitions to use for writing, instead of the definitions
+            implied by the table. Default is None, which means use the column definitions implied by the table
+        col_instructions (Optional[List[ColumnInstruction]]): instructions for customizations while writing particular
+            columns, default is None, which means no specialization for any column
+        compression_codec_name (Optional[str]): the compression codec to use. Allowed values include "UNCOMPRESSED",
+            "SNAPPY", "GZIP", "LZO", "LZ4", "LZ4_RAW", "ZSTD", etc. If not specified, defaults to "SNAPPY".
+        max_dictionary_keys (Optional[int]): the maximum number of unique keys the writer should add to a dictionary page
+            before switching to non-dictionary encoding, never evaluated for non-String columns, defaults to 2^20 (1,048,576)
+        max_dictionary_size (Optional[int]): the maximum number of bytes the writer should add to the dictionary before
+            switching to non-dictionary encoding, never evaluated for non-String columns, defaults to 2^20 (1,048,576)
+        target_page_size (Optional[int]): the target page size in bytes, if not specified, defaults to 2^20 bytes (1 MiB)
+        generate_metadata_files (Optional[bool]): whether to generate parquet _metadata and _common_metadata files,
+            defaults to False. Generating these files can help speed up reading of partitioned parquet data because these
+            files contain metadata (including schema) about the entire dataset, which can be used to skip reading some
+            files.
+        index_columns (Optional[List[List[str]]]): Lists containing the column names for indexes to persist. The write
+            operation will store the index info as sidecar tables. By default, data indexes to write are determined by
+            those present on the source table. This argument is used to narrow the set of indexes to write, or to be
+            explicit about the expected set of indexes present on all sources. Indexes that are specified but missing
+            will be computed on demand.
```

Comment:
I'm confused by the type hint and documentation.  The docs sounds like it is a list of column names (e.g. `List[str]`, but the type hint is `List[List[str]]`.
1. If this should be `List[List[str]]`, the doc should clarify what the argument contains, since I don't think I can guess with certainty.
2. I think @jmao-denver has been using `Sequence` in place of `List` in many places because it is more general.  You should discuss the `List` type hints with him.

Reply:
I have added more details to the comments explaining the argument for now. Will wait for Jianfeng to have a look regarding Sequence vs List.

Reply:
Sequence is more general and flexible. I'd prefer Sequence.